lib/kb_plant_rast/kb_plant_rastImpl.py

```diff
@@ -51,77 +55,96 @@ def annotate_plant_transcripts(self, ctx, input):
         # ctx is the context object
         # return variables are: output
         #BEGIN annotate_plant_transcripts
-
-        token = ctx['token']
-        wsClient = workspaceService(self.workspaceURL, token=token)
-
+        
         # Retrieve plant genome
-        plant_genome = wsClient.get_objects2({'objects': [{'ref' : input['input_ws']+'/'+input['input_genome']}]})['data'][0]
+        plant_genome = self.dfu.get_objects({'object_refs': [input['input_ws']+'/'+input['input_genome']]})['data'][0]
         output = {'ftrs': len(plant_genome['data']['features'])}
 
         # Retrieve kmers
-        Functions = {}
-        Kmers_Functions = {}
+        Functions = set()
+        Kmers_Functions = dict()
         for line in open('/kb/module/data/functions_kmers.txt'):
             line=line.strip()
             (function_string,kmers_string)=line.split('\t')
-            Functions[function_string]=1
+            Functions.add(function_string)
             for kmer in kmers_string.split(', '):
                 Kmers_Functions[kmer]=function_string
-        output['fns']=len(Functions.keys())
+        output['fns']=len(Functions)
+        output['kmers']=len(Kmers_Functions)
 
         Kmer_Length=8
-        Hit_Proteins={}
-        output['skipped']=0
+        Hit_Proteins=dict()
+        Hit_Kmers=set()
+        output['short']=0
         for ftr in plant_genome['data']['features']:
             Seq = ftr['protein_translation']
             SeqLen = len(Seq);
             if(SeqLen < 10):
-                output['skipped']+=1
+                output['short']+=1
                 continue
             seq_kmers = [Seq[i:i + Kmer_Length] for i in range(SeqLen-Kmer_Length+1)]
             for kmer in seq_kmers:
                 if(kmer in Kmers_Functions):
                     if(ftr['id'] not in Hit_Proteins):
-                        Hit_Proteins[ftr['id']]={}
+                        Hit_Proteins[ftr['id']]=dict()
                     if(Kmers_Functions[kmer] not in Hit_Proteins[ftr['id']]):
                         Hit_Proteins[ftr['id']][Kmers_Functions[kmer]]=0
                     Hit_Proteins[ftr['id']][Kmers_Functions[kmer]]+=1
+                    Hit_Kmers.add(kmer)
+        output['hit_kmers']=len(Hit_Kmers)
+
+        #Eliminate hits that have a small number of kmers
+        #Each function must have more than 1 kmer in order to be assigned
+        Kmer_Threshold = 1
```

Comment:
Can you imagine a case where this would be changed? I'd recommend setting this as an ALL_CAPS_GLOBAL at the top of the file or as a parameter with a defaut

Reply:
Yes I can imagine so, it's unlikely to ever happen as I'm personally going to move on from k-mers, but regardless, I'll put it in ALL_CAPS_GLOBAL